### PR TITLE
Make page titles and toc descriptions task-oriented

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Many page titles are cryptic and feature-oriented

Issue Number: N/A

## What is the new behavior?
Updates page titles and their corresponding TOC entries to be more task-oriented.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/http.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# HttpClient
+# Server interaction through HTTP
```
Comment (aikidave) [inline aio/content/guide/http.md]: I think we can make this more task oriented. Perhaps:

**Communicating with backend services using HTTP**

File: aio/content/guide/providers.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Providers
+# Provider dependencies in modules
```
Comment (aikidave) [inline aio/content/guide/providers.md]: Maybe

**Adding dependencies to modules using `providers`**

File: aio/content/guide/schematics.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Schematics
+# Code generation with schematics
```
Comment (aikidave) [inline aio/content/guide/schematics.md]: Maybe:

**Generating code using schematics**

File: aio/content/navigation.json [hunk 4]
```diff
@@ -338,8 +338,8 @@
             },
```
Comment (aikidave) [inline aio/content/navigation.json]: Please update with the modified headings.

File: aio/content/guide/providers.md (hunk 2 above)
Comment (jbogarthyde) [reply aio/content/guide/providers.md]: Mistake -- This was supposed to be "Providing dependencies in modules"

**Review by aikithoughts — APPROVED** [review]

**Pull request closed without merging**